test/features/articles/creating_an_article_test.rb

```diff
@@ -1,15 +1,66 @@
 require "test_helper"
 
 feature "Creating an Article" do
-  scenario "submit form data to create a new article" do
-    article = articles(:two)
+
+  scenario "returns success if user is an author" do
+    sign_in(:wilbur)
+    create_article
+    page.must_have_content "Article was successfully created."
+    page.must_have_content @article.title
+    page.must_have_content @article.body
+  end
+
+  scenario "returns success if user is the editor" do
+    sign_in(:editor)
+    create_article
+    page.must_have_content "Article was successfully created."
+    page.must_have_content @article.title
+    page.must_have_content @article.body
+  end
+
+  scenario "must show author name on article page" do
+    sign_in(:wilbur)
+    create_article
+    page.has_css? "#author"
+    page.text.must_include @user.email
+  end
+
+  scenario "must show status as unpublished" do
+    sign_in(:wilbur)
+    create_article
+    page.text.must_include "Status: Unpublished"
+  end
+
+  scenario "returns error message for visitors" do
     visit new_article_path
-    page.must_have_content "New Article"
-    fill_in "Title", with: article.title
-    fill_in "Body", with: article.body
+    page.must_have_content "You need to sign in or sign up before continuing"
+  end
+
+  scenario "button will not appear for visitors" do
+    visit articles_path
+    page.wont_have_link "New Article"
+  end
+
+  scenario "redirects unauthorized visitors" do
+    sign_in(:benji)
+    visit new_article_path
+    page.text.must_include "You are not authorized to perform this action."
+  end
+
+  scenario "authors can't publish" do
+    sign_in(:wilbur)
+    visit new_article_path
+    page.wont_have_field('Published')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.